examples/04_New Models.ipynb

```diff
@@ -1393,12 +1393,261 @@
+    "    else:\n",
+    "        i = 4\n",
+    "    return batt.InputContainer({'i': i})"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Now, let's simulate each model to threshold using the `simulate_to_threshold` method. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "# Original model \n",
+    "results_orig = batt.simulate_to_threshold(future_loading, save_freq = 1)\n",
+    "# Pickled version  \n",
+    "results_pkl = load_pkl.simulate_to_threshold(future_loading, save_freq = 1)\n",
+    "# JSON versions\n",
+    "results_json_1 = json_1.simulate_to_threshold(future_loading, save_freq = 1)\n",
+    "results_json_2 = json_2.simulate_to_threshold(future_loading, save_freq = 1)\n"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Finally, let's plot the results for comparison."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "voltage_orig = [results_orig.outputs[iter]['v'] for iter in range(len(results_orig.times))]\n",
+    "voltage_pkl = [results_pkl.outputs[iter]['v'] for iter in range(len(results_pkl.times))]\n",
+    "voltage_json_1 = [results_json_1.outputs[iter]['v'] for iter in range(len(results_json_1.times))]\n",
+    "voltage_json_2 = [results_json_2.outputs[iter]['v'] for iter in range(len(results_json_2.times))]\n",
+    "\n",
+    "plt.plot(results_orig.times,voltage_orig,'-b',label='Original surrogate') \n",
+    "plt.plot(results_pkl.times,voltage_pkl,'--r',label='Pickled serialized surrogate') \n",
+    "plt.plot(results_json_1.times,voltage_json_1,'-.g',label='First JSON serialized surrogate') \n",
+    "plt.plot(results_json_2.times, voltage_json_2, '--y', label='Second JSON serialized surrogate')\n",
+    "plt.legend()\n",
+    "plt.xlabel('Time (sec)')\n",
+    "plt.ylabel('Voltage (volts)')"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "All of the voltage curves overlap, showing that the different serialization methods produce the same results. \n",
+    "\n",
+    "Additionally, we can compare the output arrays directly, to ensure equivalence. "
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import numpy as np\n",
+    "\n",
+    "# Check if the arrays are the same\n",
+    "are_arrays_same = np.array_equal(voltage_orig, voltage_pkl) and \\\n",
+    "                  np.array_equal(voltage_orig, voltage_json_1) and \\\n",
+    "                  np.array_equal(voltage_orig, voltage_json_2)\n",
+    "\n",
+    "print(f\"The simulated results from the original and serialized models are {'identical. This means that our serialization works!' if are_arrays_same else 'not identical. This means that our serialization does not work.'}!\")"
```

Comment:
Why is this line in the python instead of a markdown cell?

Reply:
Aditya added this in his notebook. He checks that the output arrays are the same, and what is printed in this statement is dependent on if are_arrays_same evaluates to True. 

I see why it's a bit silly, since we've set up this example so that this will always be the case. I personally prefer a visual of the results, which we have above. But Aditya added this because in the plot, all the lines lie on top of each other, so it's a bit confusing and hard to see. 

Thoughts on if we should include this or get rid of it, @teubert?

Reply:
Ah- I see now. That's fine, let's keep it.